File: Data Structures & Algorithms/validate-parentheses/submission-6.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        Map<Character, Character> close_to_open = new HashMap<>();
        close_to_open.put(')', '(');
        close_to_open.put('}', '{');
        close_to_open.put(']', '[');
        for (char c: s.toCharArray()) {
            if (close_to_open.containsKey(c)) {
                if (!stack.isEmpty() && stack.peek()==close_to_open.get(c)) {
                    stack.pop();
                }
                else {
                    return false;
                }
            }
            else {
                stack.push(c);
            }
        }
        return stack.isEmpty();
    }
}

File: Data Structures & Algorithms/is-anagram/submission-6.cpp
class Solution {
    
public:
    bool isAnagram(string s, string t) {
        vector<int> hashMap(26, 0);
        for (const char c:s) {
            hashMap[c - 'a']++;
        }

        for (const char c:t) {
            if (hashMap[c - 'a']) hashMap[c - 'a']--;
            else return false;
        }

        for (const int n: hashMap) {
            if (n != 0) return false;
        }

        return true;
    }
};
